lightkurve/correctors/designmatrix.py

```diff
@@ -231,28 +242,39 @@ def append_constant(self, prior_mu=0, prior_sigma=np.inf):
         return DesignMatrix(new_df, name=self.name,
                             prior_mu=prior_mu, prior_sigma=prior_sigma)
 
-    def _validate(self):
-        """Raises a `LightkurveWarning` if the matrix has a low rank."""
+    def _validate(self, rank=True):
```

Comment:
Yes, we need to ignore the rank warning when we use sparse. Calculating the rank warning requires we evaluate the whole matrix (using scipy.sparse.toarray) which completely negates the value of using sparse matrices. I did look into a sparse way of calculating rank, but i think it can't be done.

Reply:
That makes sense!  May be worth putting this in the docstring.  It would prevent people like me activating the rank warning in the future without thinking!